metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -893,7 +893,7 @@ def emit_custom_sql_datasources(self) -> Iterable[MetadataWorkUnit]:
             view_properties = ViewPropertiesClass(
                 materialized=False,
                 viewLanguage="SQL",
-                viewLogic=clean_query(csql.get("query", "")),
+                viewLogic=clean_query(csql.get("query") or ""),
```

Comment:
Suggestion: Let's just skip adding viewProperties to dataset aspects if custom SQL query is not available.